components/application-mgt/org.wso2.carbon.device.application.mgt.api/src/main/java/org/wso2/carbon/device/application/mgt/api/APIUtil.java

```diff
@@ -232,4 +226,22 @@ public static SubscriptionManager getSubscriptionManager() {
 
         return subscriptionManager;
     }
+
+    public static CommentsManager getCommentsManager() {
+        if (commentsManager == null) {
+            synchronized (APIUtil.class) {
+                if (commentsManager == null) {
+                    PrivilegedCarbonContext ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+                    commentsManager =
+                            (CommentsManager) ctx.getOSGiService(CommentsManager.class, null);
```

Comment:
Don't cache `CommentsManager` into variable `commentsManager`. By nature OSGi services may come and go; thus your references sooner will be outdated. Always get `commentsManager` through `ctx.getOSGiService` call.